src/meltano/cli/lock.py

```diff
@@ -77,6 +77,10 @@ def lock(
 
     tracked_plugins = []
 
+    if not plugins:
+        raise CliError("No matching plugin(s) found")
```

Comment:
1. We may want to track the event (@WillDaSilva can confirm)
2. Errors are usually easier to read when the string is not repeated in the traceback and the final output message

```suggestion
        tracker.track_command_event(CliEvent.aborted)
        errmsg = "No matching plugin(s) found"
        raise CliError(errmsg)
```

Reply:
@edgarrmondragon thanks! Based on Will's thumbs up I assumed he liked your suggestion so I went ahead and pulled it in.